joeynmt/search.py

```diff
@@ -328,10 +329,13 @@ def beam_search(
                 finished_hyp = is_finished[i].nonzero().view(-1)
                 # store finished hypotheses for this batch
                 for j in finished_hyp:
-                    hypotheses[b].append((
-                        topk_scores[i, j],
-                        predictions[i, j, 1:])  # ignore start_token
-                    )
+                    # Check if the prediction has more than one EOS.
+                    # If it has more than one EOS, it means that the prediction should have already
+                    # been added to the hypotheses, so you don't have to add them again.
```

Comment:
Thanks for adding comments about this as well, this is will help others to understand what's going on :+1: